Sources/NIOConcurrencyHelpers/lock.swift

```diff
@@ -21,6 +21,20 @@ import WinSDK
 import Glibc
 #endif
 
+#if !os(Windows)
```

Comment:
much like for `Selector*.swift`, I'd suggest to split this into `LockPthread.swift`/`LockOSUnfairLock.swift`/`LockWindows.swift` etc. Then using one big `#if os(...)` at the top of those files supply the right one for the right platform (alongside an `internal LockProtocol` protocol that's just there to set the API and is never used.